test/test_serialization.py

```diff
@@ -0,0 +1,314 @@
+            (iterdp.ParagraphAggregator, (), {}),
+            # (iterdp.ParquetDataFrameLoader, (), {"dtype": DTYPE}),
+            (iterdp.RarArchiveLoader, (), {}),
+            (iterdp.Rows2Columnar, (), {}),
+            (iterdp.SampleMultiplexer, (), {}),
+            (iterdp.Saver, (), {"mode": "wb", "filepath_fn": partial(_filepath_fn, dir=self.temp_dir.name)}),
+            (iterdp.TarArchiveLoader, (), {}),
+            (iterdp.UnZipper, (), {"sequence_length": 2}),
+            (iterdp.XzFileLoader, (), {}),
+            (iterdp.ZipArchiveLoader, (), {}),
+        ]
+
+        picklable_datapipes = _filter_by_module_availability(picklable_datapipes)
+
+        # Custom input_dp for when `IterableWrapper(range(10))` does not work for the DataPipe being tested
+        custom_input = {
+            iterdp.BucketBatcher: IterableWrapper([0, 0, 0, 0, 0, 0, 0]),
+            iterdp.CSVDictParser: IterableWrapper(
+                [("f1", StringIO("Label,1,1\nLabel,2,2\nLabel,3,3")), ("f2", StringIO("L,1,1\r\nL,2,2\r\nL,3,3"))]
+            ),
+            iterdp.CSVParser: IterableWrapper(
+                [("f1", StringIO("Label,1,1\nLabel,2,2\nLabel,3,3")), ("f2", StringIO("L,1,1\r\nL,2,2\r\nL,3,3"))]
+            ),
+            iterdp.DataFrameMaker: IterableWrapper([(i,) for i in range(3)]),
+            iterdp.FSSpecFileLister: ".",
+            iterdp.FSSpecSaver: IterableWrapper([("1.txt", b"DATA1"), ("2.txt", b"DATA2"), ("3.txt", b"DATA3")]),
+            iterdp.IndexAdder: IterableWrapper([{"a": 1, "b": 2}, {"c": 3, "a": 1}]),
+            iterdp.IoPathFileLister: ".",
+            iterdp.IoPathSaver: IterableWrapper([("1.txt", b"DATA1"), ("2.txt", b"DATA2"), ("3.txt", b"DATA3")]),
+            iterdp.IterKeyZipper: IterableWrapper([("a", 100), ("b", 200), ("c", 300)]),
+            iterdp.JsonParser: IterableWrapper(
+                [
+                    ("1.json", StringIO('["fo", {"ba":["baz", null, 1.0, 2]}]')),
+                    ("2.json", StringIO('{"__cx__": true, "r": 1, "i": 2}')),
+                ]
+            ),
+            iterdp.LineReader: IterableWrapper(
+                [("file1", StringIO("Line1\nLine2")), ("file2", StringIO("Line2,1\r\nLine2,2\r\nLine2,3"))]
+            ),
+            iterdp.MapKeyZipper: IterableWrapper([("a", 1), ("b", 2), ("c", 3)]),
+            iterdp.ParagraphAggregator: IterableWrapper([("f1", "L1"), ("f1", "L2"), ("f2", "21"), ("f2", "22")]),
+            iterdp.Rows2Columnar: IterableWrapper([[{"a": 1}, {"b": 2, "a": 1}], [{"a": 1, "b": 200}, {"c": 3}]]),
+            iterdp.SampleMultiplexer: {IterableWrapper([0] * 10): 0.5, IterableWrapper([1] * 10): 0.5},
+            iterdp.Saver: IterableWrapper([("1.txt", b"DATA1"), ("2.txt", b"DATA2"), ("3.txt", b"DATA3")]),
+            iterdp.UnZipper: IterableWrapper([(i, i + 10) for i in range(10)]),
+        }
+
+        # Skipping value comparison for these DataPipes
+        dp_skip_comparison = {
```

Comment:
Then, it looks good to me. I would love to having a few comments about why we skip them here.

Reply:
Yeah because TorchArrow column has `==` overloaded (similar to Pandas/PyTorch). So in `self.assertEqual`, it did something like 
```
c1 == c2 -> ta.Column([False,...])
self.assert(first,second) -> if not first == second
                          -> if not ta.Column([False,...]) == [True, False, ...]
```

So `first == second` will be evaluated into a boolean column. And `not first == second` will always returns True. 

One way is to do something like 
```
self.assertTrue((c1 == c2).all())
```

See https://github.com/facebookresearch/torcharrow/issues/238 context.

Reply:
This is really good to know. Thanks!